resolwe_runtime_utils.py

```diff
@@ -275,14 +301,224 @@ def checkrc(rc, *args):
     return json.dumps(ret)
 
 
+def export_file(file_path):
+    """Prepend the given parameter with ``export``"""
+
+    if not os.path.isfile(file_path):
+        return error("Referenced file does not exist: '{}'.".format(file_path))
+
+    return "export {}".format(file_path)
+
+
+CHUNK_SIZE = 10000000  # 10 Mbytes
+
+
+class ImportedFormat:
+    """Import destination file format."""
+
+    EXTRACTED = 'extracted'
+    COMPRESSED = 'compressed'
+    BOTH = 'both'
+
+
+def import_file(
+    src,
+    file_name,
+    imported_format=ImportedFormat.BOTH,
+    progress_from=0.0,
+    progress_to=None,
+):
+    """Import file to working directory.
+
+    :param src: Source file path or URL
+    :param file_name: Source file name
+    :param imported_format: Import file format (extracted, compressed or both)
+    :param progress_from: Initial progress value
+    :param progress_to: Final progress value
+    :return: Destination file path (if extracted and compressed, extracted path given)
+    """
+
+    if progress_to is not None:
+        if not isinstance(progress_from, float) or not isinstance(progress_to, float):
+            raise ValueError("progress_from and progress_to must be float")
+
+        if progress_from >= progress_to:
+            raise ValueError("progress_to must be higher than progress_from")
+
+    print("Importing and compressing {}...".format(file_name))
+
+    def importGz():
+        """Import gzipped file.
+
+        The file_name must have .gz extension.
+        """
+        if imported_format != ImportedFormat.EXTRACTED:  # Compressed file required
+            with gzip.open(src, 'rb') as f:
+                try:
+                    while f.read(CHUNK_SIZE) != b'':
+                        pass
+                except Exception:
+                    print(error("Invalid gzip file format"))
+            try:
+                shutil.copyfile(src, file_name)
+            except shutil.SameFileError:
+                pass  # Skip copy of downloaded files
+
+        if imported_format != ImportedFormat.COMPRESSED:  # Extracted file required
+            with open(file_name[:-3], 'wb') as f_out, gzip.open(src, 'rb') as f_in:
+                shutil.copyfileobj(f_in, f_out, CHUNK_SIZE)
+
+        return (
+            file_name
+            if imported_format == ImportedFormat.COMPRESSED
+            else file_name[:-3]
+        )
+
+    def import7z():
+        """Import compressed file in various formats.
+
+        Supported extensions: .bz2, .zip, .rar, .7z, .tar.gz, and .tar.bz2.
+        """
+        extracted_name, _ = os.path.splitext(file_name)
+        temp_dir = 'temp_{}'.format(extracted_name)
```

Comment:
This is highly unlikely because the name is `temp_<file_name>`. Also, the temporary directory is removed after upload so another field's re-import can not cause this error. 

We have a greater problem when users would upload files of the same name which is problematic in all use cases. However, there are no known occurrences of this issue.